runtime/packages/core-modules/assembly/system.ts

```diff
@@ -0,0 +1,63 @@
+import { Hash, CompactInt, ByteArray, Bytes } from 'as-scale-codec';
+import { Storage } from './storage';
+import { Header } from '@as-substrate/models';
+import { Helpers } from './helpers';
+
+export class System {
+    // execution phases
+    static readonly APPLY_EXTRINSIC: string = "ApplyExtrinsic";
+    static readonly INITIALIZATION: string = "Initialization";
+    static readonly FINALIZATION: string = "Finalization"
+    /**
+     * Sets up the environment necessary for block production
+     * @param header Header instance
+     */
+    static initialize(header: u8[]): void{
```

Comment:
Instead of parsing the header, why don't you instanciate the `Header` class from the bytes?

Reply:
well, I thought it would be redundant. First converting input bytes to header instance, then converting header back to bytes when I set them to the storage seemed unnecessary, that's why I am just slicing the bytes and setting them to the storage directly. I could create some constant variables for the well-known sizes